sdk/storage/azure-storage-file-share/azure/storage/fileshare/_generated/aio/operations_async/_share_operations_async.py

```diff
@@ -395,8 +395,7 @@ def __init__(self, client, config, serializer, deserializer) -> None:
         # Construct headers
         header_parameters = {}
         header_parameters['Accept'] = 'application/json'
-        if file_permission_key is not None:
-            header_parameters['x-ms-file-permission-key'] = self._serialize.header("file_permission_key", file_permission_key, 'str')
+        header_parameters['x-ms-file-permission-key'] = self._serialize.header("file_permission_key", file_permission_key, 'str')
```

Comment:
Breaking change: please check our surface API was already making this guy required, or at least was always fitting a default if not provided by customer

Reply:
yes, it's required